src/database/repositories/users.ts

```diff
@@ -19,14 +20,29 @@ export const addUser = async (user: user): Promise<void> => {
 
 export const getUserByUsernamePassword = async (
   username: string,
-  password: string
+  hashedPassword: string
 ): Promise<user | null> => {
   const result = await db
     .select()
     .from(users)
-    .where(and(eq(users.username, username), eq(users.password, password)))
+    .where(and(eq(users.username, username), eq(users.password, hashedPassword)))
+  if (result.length < 1) {
+    return null
```

Comment:
again, not my preference returning null. Throwing errors and catching them is usually a better pattern